src/backend/wsServer.ts

```diff
@@ -75,8 +75,10 @@ function hasPermission(playerId: PlayerId, gameId: GameId, gameEvent: GameEvent)
 		// to power to increase or decrease scores is the server
 		// so we may change this later
 		return playerId === gameEvent.data.id;
+	} else if (gameEvent.type === 'change-player-name') {
```

Comment:
add an `if` check for the new event type for the `hasPermission` function. the role of the `hasPermission` function is to check if the player who produced this game event has permission to do so, e.g. a player should only be able to change their own name, not change the name of other players.